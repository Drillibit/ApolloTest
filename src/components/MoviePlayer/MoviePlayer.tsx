import React, { PureComponent, Fragment } from 'react';
import styled from 'styled-components';

import { Icon } from '../UIKit/Icon';
import { Button } from '../UIKit/Button';

const VideoWrapper = styled.div`
  color: black;
  position: relative;
  height: 0;
  width: 100%;
  border-radius: 2px;
`;

const StyledIframe = styled.iframe`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 284px;
  z-index: 1;
  border-radius: 2px;
`;

const StyledPreload = styled.div`
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  top:0;
  left: 0;
  width: 100%;
  height: 284px;
  background-color: black;
  border-radius: 2px;
`;
type StyledOveralyType = {
  bg: string
}
const StyledOveraly = styled.div`
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  top:0;
  left: 0;
  width: 100%;
  height: 284px;
  background-color: black;
  background-image: 
   url('${({ bg }:StyledOveralyType) => `https://image.tmdb.org/t/p/original${bg}`}');
  background-size: cover;
  z-index: 10;
  border-radius: 2px;
`;
export type MoviePlayerProps = {
  link: string,
  image: string,
  playing: boolean,
  onPlay: () => void
}

export class MoviePlayer extends PureComponent<MoviePlayerProps> {
  render() {
    const {
      link,
      image,
      playing,
      onPlay
    } = this.props;

    return (
      <VideoWrapper>
        {!playing && (
          <StyledOveraly bg={image}>
            <Button btnType="primary" btnSize="middle" onClick={onPlay}>
              <Icon size="sm" color="#fff" icon="play" />
              Смотреть трейлер
            </Button>
          </StyledOveraly>
        )}
        {playing && (
          <Fragment>
            <StyledPreload />
            <StyledIframe
              id="ytplayer"
              src={`http://www.youtube.com/embed/${link}?autoplay=1`}
              frameBorder="0"
              allowFullScreen
            />
          </Fragment>
        )}
      </VideoWrapper>
    );
  }
}