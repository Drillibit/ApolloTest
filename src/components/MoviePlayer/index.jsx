import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const VideWrapper = styled.div`
  position: relative;   
  padding-bottom: 20rem; 
  height: 0;
`;

const StyledIframe = styled.iframe`
  position: absolute;
  top:0;
  left: 0;
  width: 100%;
  height: 100%;
`;

export const MoviePlayer = ({ id }) => (
  <VideWrapper>
    <StyledIframe
      title={id}
      key={id}
      id="ytplayer"
      type="text/html"
      src="http://www.youtube.com/embed/_xBziiJI3DU?1"
      frameBorder="0"
    />
  </VideWrapper>);

MoviePlayer.propTypes = {
  id: PropTypes.number.isRequired
};
