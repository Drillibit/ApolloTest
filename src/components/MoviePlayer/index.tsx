import React, { PureComponent, Fragment } from 'react';
import styled from 'styled-components';
import { string, bool, func, } from 'prop-types';

import { Icon } from '../UIKit/Icon';
import { Button } from '../UIKit/Button';

const VideoWrapper = styled.div`
  color: black;
  position: relative;
  height: 0;
  width: 100%;
  border-radius: 2px;
`;

const StyledIframe = styled.iframe`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 284px;
  z-index: 1;
  border-radius: 2px;
`;

const StyledPreload = styled.div`
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  top:0;
  left: 0;
  width: 100%;
  height: 284px;
  background-color: black;
  border-radius: 2px;
`;

const StyledOveraly = styled.div`
  position: absolute;
  display: flex;
  align-items: center;
  justify-content: center;
  top:0;
  left: 0;
  width: 100%;
  height: 284px;
  background-color: black;
  background-image: 
   url('${({ bg }) => `https://image.tmdb.org/t/p/original${bg}`}');
  background-size: cover;
  z-index: 10;
  border-radius: 2px;
`;

export class MoviePlayer extends PureComponent {
  render() {
    const {
      link,
      image,
      playing,
      onPlay
    } = this.props;

    return (
      <VideoWrapper>
        {!playing && (
          <StyledOveraly bg={image}>
            <Button btnType="primary" btnSize="middle" onClick={onPlay}>
              <Icon icon="play" />
              Смотреть трейлер
            </Button>
          </StyledOveraly>
        )}
        {playing && (
          <Fragment>
            <StyledPreload />
            <StyledIframe
              id="ytplayer"
              type="text/html"
              src={`http://www.youtube.com/embed/${link}?autoplay=1`}
              frameBorder="0"
              allowFullScreen="1"
            />
          </Fragment>
        )}
      </VideoWrapper>
    );
  }
}

MoviePlayer.propTypes = {
  link: string,
  image: string,
  playing: bool,
  onPlay: func
};

MoviePlayer.defaultProps = {
  link: '/',
  image: '/',
  playing: false,
  onPlay: f => f
};
