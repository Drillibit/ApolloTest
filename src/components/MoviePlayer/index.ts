import { MoviePlayer } from './MoviePlayer';

export {
    MoviePlayer
}
