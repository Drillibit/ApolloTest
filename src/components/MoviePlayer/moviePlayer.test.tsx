import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { MoviePlayer, MoviePlayerProps } from './MoviePlayer';

const fakePlayerData:MoviePlayerProps = {
    link: 'safdfsdlm',
    image: 'sdfdsffsg',
    playing: false,
    onPlay: jest.fn()
};

describe('<MoviePlayer />', () => {
    it('renders player component', () => {
        const wrapper = shallow(<MoviePlayer {...fakePlayerData} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    })
}); 
