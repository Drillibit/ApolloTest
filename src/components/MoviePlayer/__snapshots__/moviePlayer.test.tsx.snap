// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MoviePlayer /> renders player component 1`] = `
<styled.div>
  <styled.div
    bg="sdfdsffsg"
  >
    <Component
      btnSize="middle"
      btnType="primary"
      onClick={[MockFunction]}
    >
      <Icon
        color="#fff"
        icon="play"
        size="sm"
      />
      Смотреть трейлер
    </Component>
  </styled.div>
</styled.div>
`;
